WcaOnRails/app/views/registrations/register.html.erb

```diff
@@ -42,7 +42,7 @@
     <% end %>
   <% else %>
 
-    <% if @competition.registration_opened? %>
+    <% if @competition.registration_opened? && @competition.showAtAll? %>
```

Comment:
I had added some code in the controller to stop the organiser from registering. I get what you are saying though and had some similar thoughts myself in that an organiser should be able to register. As the competition warnings are already displayed on the page there is now a warning that says "This competition is not visible to the public". So I think I can just add a test showing that a user will get a 'Not Found' exception, and that an organiser can register? I'll put this together soon in a new PR so people can comment.